dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/BitBucketSourceCodeRepo.java

```diff
@@ -279,46 +159,71 @@ public String getOrganizationEmail() {
         return reposByGitURl;
     }
 
-    @Override public Workflow getNewWorkflow(String repositoryId, Optional<Workflow> existingWorkflow) {
-        // repository id of the form owner/name
+    /**
+     * Uses Bitbucket API to grab a raw source file and return it
+     * @param path
+     * @param repositoryId
+     * @param branch
+         * @param type
+         * @return source file
+         */
+    private SourceFile getSourceFile(String path, String repositoryId, String branch, SourceFile.FileType type) {
+        // TODO: should we even be creating a sourcefile before checking that it is valid?
+        // I think it is fine since in the next part we just check that source file has content or not (no content is like null)
+        SourceFile file = new SourceFile();
+
+        // Get descriptor content using the BitBucket API
+        String url = BITBUCKET_API_URL + "repositories/" + repositoryId + "/raw/" + branch + "/" + path;
+        Optional<String> asString = ResourceUtilities.asString(url, bitbucketTokenContent, client);
+
+        LOG.info(gitUsername + ": RESOURCE CALL: {}", url);
+
+        if (asString.isPresent()) {
+            // Grab content from found file
+            String content = asString.get();
+
+            // Is workflow descriptor valid?
+            Boolean validWorkflow;
```

Comment:
No reason, was a misunderstanding of the differences on my part. Will change to boolean